packages/vitest/src/node/workspace.ts

```diff
@@ -85,6 +86,14 @@ export class WorkspaceProject {
     return this.ctx.getCoreWorkspaceProject() === this
   }
 
+  provide = (key: string, value: unknown) => {
```

Comment:
Should a warning be added if `value` is not serializable?
A simple one might be `strictEqual(JSON.parse(JSON.stringify(value)), value)`.

Also, it could help adding `SerializableValue` type to help user see complex structure will not work (though the calculation might be slow).

Reply:
No, serialization depends on the pool used. It's not json.stringify.

Reply:
Should user be warned in case their data is not serializable?

Reply:
Vitest will throw an error during serialization

Reply:
I guess we can technically check with `structureEqual`

Reply:
> Vitest will throw an error during serialization

Oh, so the user will be able to get a hint on what's gone wrong?
Can the test for that be added?